crates/iota/src/fire_drill.rs

```diff
@@ -161,23 +160,20 @@ async fn update_next_epoch_metadata(
     new_config.protocol_key_pair =
         KeyPairWithPath::new(IotaKeyPair::Ed25519(new_protocol_key_pair));
 
-    let iota_system_state = iota_client
+    // needs to be active_validators instead of committee_members here, so that
+    // every validator can update their own metadata
+    let self_active_validator = iota_client
         .governance_api()
         .get_latest_iota_system_state()
-        .await?;
-    let validators = match iota_system_state {
-        IotaSystemStateSummary::V1(v1) => v1.active_validators,
-        IotaSystemStateSummary::V2(v2) => v2.active_validators,
-        _ => panic!("unsupported IotaSystemStateSummary"),
-    };
-
-    let self_validator = validators
-        .iter()
+        .await?
+        .iter_active_validators()
         .find(|v| v.iota_address == iota_address)
-        .unwrap();
+        .unwrap()
```

Comment:
Is this always safe? Please check also all other unwraps in the PR.

Reply:
Agreed, modified all the non-test related codes touched in this PR in [91630f2](https://github.com/iotaledger/iota/pull/6015/commits/91630f2f4dab02f4aa17ea504b300075e252b2a5).

Reply:
Awesome, thank you!